resources/js/beatmap-discussions/love-beatmap-modal.tsx

```diff
@@ -99,32 +106,36 @@ export default class LoveConfirmation extends React.Component<Props> {
     const mode = e.target.value as GameMode;
     const modeBeatmaps = this.groupedBeatmaps.get(mode) ?? [];
 
-    const action = this.checkIsModeSelected(mode) ? 'delete' : 'add';
-    modeBeatmaps.forEach((beatmap) => this.selectedBeatmapIds[action](beatmap.id));
+    const op = this.checkIsModeSelected(mode) ? 'delete' : 'add';
+    modeBeatmaps.forEach((beatmap) => this.selectedBeatmapIds[op](beatmap.id));
   };
 
+  @action
   private handleSubmit = () => {
-    if (!confirm(trans('beatmaps.nominations.love_confirm'))) {
-      return;
-    }
-
-    if (this.selectedBeatmapIds.size === 0) {
+    if (this.xhr != null
+      || this.selectedBeatmapIds.size === 0
+      || !confirm(trans('beatmaps.nominations.love_confirm'))) {
       return;
     }
 
     showLoadingOverlay();
+    showLoadingOverlay.flush();
```

Comment:
isn't the disabling the button enough? Seems inconsistent with other places 🤔 Otherwise there's `showImmediateLoadingOverlay`

Reply:
I'm not sure why it previously even had the overlay since it didn't prevent anything until the timer expires 👀 
Immediately showing the overload at least prevents the modal from being closed while the request is pending; the other option is to disable closing the modal.